File: BotTemplate/BotTemplate-Java/src/main/java/io/github/freya022/bot/config/Environment.java
package io.github.freya022.bot.config;

import java.nio.file.Files;
import java.nio.file.Path;

public class Environment {
    /**
     * The mode is determined by checking if the
     * {@code dev-config} directory exists in the current directory.
     */
    public static final boolean IS_DEV = Files.exists(Path.of("dev-config"));

    /**
     * The folder where the data and configuration directories reside.
     * <br>This is the current <b>working directory</b>.
     */
    public static final Path FOLDER = Path.of("");

    public static final Path CONFIG_FOLDER = FOLDER.resolve(IS_DEV ? "dev-config" : "config");
    public static final Path LOGBACK_CONFIG_PATH = CONFIG_FOLDER.resolve(IS_DEV ? "logback-test.xml" : "logback.xml");
}

File: src/examples/kotlin/io/github/freya022/bot/config/Environment.kt
package io.github.freya022.bot.config

import java.nio.file.Path
import kotlin.io.path.Path
import kotlin.io.path.exists

object Environment {
    /**
     * The folder where the data and configuration directories reside.
     */
    val folder: Path = Path("test-files", "examples")

    /**
     * The mode is determined by checking if the
     * `dev-config` directory exists in the current directory.
     */
    val isDev: Boolean = folder.resolve("dev-config").exists()

    val configFolder: Path =
        folder.resolve(if (isDev) "dev-config" else "config")
    val logbackConfigPath: Path = configFolder.resolve(if (isDev) "logback-test.xml" else "logback.xml")
}